assembl/views/api2/auth.py

```diff
@@ -368,31 +368,30 @@ def reset_password(request):
                     localizer.translate(_("This email does not exist")),
                     code=HTTPNotFound.code)
             else:
-                raise JSONError(
-                    localizer.translate(_(generic_error_message)),
-                    code=HTTPNotFound.code)
+                send_change_password_email(request, user, email, discussion=discussion)
                 logger.error("This email does not exist.")
         if account:
             email = account.email
     else:
         error = localizer.translate(_("Please give an identifier"))
         raise JSONError(error)
     if not email:
-        email = user.get_preferred_email()
+        if user:
+            email = user.get_preferred_email()
+        else:
+            email = None
```

Comment:
Could be refactored to:
```python
if user and not email:
    email = user.get_preferred_email()
```

Reply:
If you `return` early line 371, you shouldn't need this change.